burnman/nonlinear_fitting.py

```diff
@@ -0,0 +1,314 @@
+import numpy as np
+from scipy.stats import t
+import itertools
+    
+def nonlinear_least_squares_fit(model,
+                                mle_tolerance,
+                                lm_damping = 0.,
+                                param_tolerance = 1.e-7,
+                                max_lm_iterations = 100,
+                                verbose = False):
+
+    """
+    Function to compute the "best-fit" parameters for a model
```

Comment:
Yes, this is still L-M. The new function runs a full nonlinear inversion with covariances in all parameters, rather than an inversion with errors only in one direction ("y"). No particular coordinate axis is special in the new formulation, so from that perspective it's a bit like orthogonal distance regression.